src/ConstructiveSolidGeometry/VolumePrimitives/Torus.jl

```diff
@@ -1,89 +1,130 @@
-struct Torus{T,TR,TB,TP,TT} <: AbstractVolumePrimitive{T}
+struct Torus{T,TR,TB,TP,TT,TZ} <: AbstractVolumePrimitive{T}
```

Comment:
I'm not sure, whether we need that `TZ`, because `z` is **ALWAYS** of type `T`

Reply:
This also happens in many other places. Like for TR. And then for the same fields in surface primitives. Should we keep these parametric types as short as possible or tie the order and number of parameters to the number of fields in the struct?

Reply:
That's a good question! @lmh91 what's your opinion on this?

Reply:
If it is really **always** of type `T` get rid of it. 

Parametric types are nice but should be kept as small as possible.